api/provider/aws/service_update.go

```diff
@@ -0,0 +1,85 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (s *ServiceProvider) Update(req models.UpdateServiceRequest) error {
+	serviceID := req.ServiceID
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
+
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	serviceName := fqServiceID
+
+	if req.DeployID != nil {
+		deployID := *req.DeployID
+		taskDefinitionFamily, err := lookupTaskDefinitionFamily(s.TagStore, deployID)
+		if err != nil {
+			return err
+		}
+
+		taskDefinitionRevision, err := lookupTaskDefinitionRevision(s.TagStore, deployID)
+		if err != nil {
+			return err
+		}
+
+		if err := s.updateServiceTaskDefinition(clusterName, serviceName, taskDefinitionFamily, taskDefinitionRevision); err != nil {
+			return err
+		}
+	}
+
+	if req.Scale != nil {
+		desiredCount := *req.Scale
+		if err := s.updateServiceDesiredCount(clusterName, serviceName, desiredCount); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) updateServiceTaskDefinition(clusterName, serviceName, taskDefinitionFamily, taskDefinitionRevision string) error {
+	input := &ecs.UpdateServiceInput{}
+	input.SetCluster(clusterName)
+	input.SetService(serviceName)
+
+	taskDefinition := fmt.Sprintf("%s:%d", taskDefinitionFamily, taskDefinitionRevision)
```

Comment:
`taskDefintionRevision` is a string, so this should be `%s:%s`

Reply:
✅